### PR TITLE
feat: add support for Laravel 8 and upper tests

### PR DESCRIPTION
**Pull request by ThibaudDauce**
**Changes**

The new `tests/Features/Laravel8` folder can contains tests only executed on Laravel 8 and above.

Needed to add the `sole` method support https://github.com/nunomaduro/larastan/pull/820#issuecomment-824741968

### CODE REVIEW

File: tests/FeaturesTest.php [hunk 1]
```diff
@@ -37,6 +38,12 @@ public function testFeatures(string $file): void
 
     private function analyze(string $file): int
     {
+        $laravelMajorVersion = (int) explode('.', app()->version())[0];
+
+        if (Str::contains($file, 'Features/Laravel8') && $laravelMajorVersion < 8) {
```
Comment (szepeviktor) [inline tests/FeaturesTest.php:43]: Please consider using `compare_version()` with < 8.0.0

Comment (canvural) [reply tests/FeaturesTest.php]: I agree with @szepeviktor

File: tests/FeaturesTest.php [hunk 2]
```diff
@@ -37,6 +38,12 @@ public function testFeatures(string $file): void
 
     private function analyze(string $file): int
     {
+        $laravelMajorVersion = (int) explode('.', app()->version())[0];
```
Comment (canvural) [inline tests/FeaturesTest.php]: I would rather have these checks in `getFeatures` method.

**Review by canvural — CHANGES_REQUESTED** [review]

File: tests/FeaturesTest.php (hunk 1 above)
Comment (ThibaudDauce) [reply tests/FeaturesTest.php]: Nice! Didn't know this PHP function :-D

File: tests/FeaturesTest.php (hunk 2 above)
Comment (ThibaudDauce) [reply tests/FeaturesTest.php]: It's not possible because the framework is not initialize yet in a data provider (I think it's the reason)

Comment (canvural) [reply tests/FeaturesTest.php]: Ok, good point! So let's leave it for now :+1:

File: tests/FeaturesTest.php [hunk 3]
```diff
@@ -37,6 +38,10 @@ public function testFeatures(string $file): void
 
     private function analyze(string $file): int
     {
+        if (Str::contains($file, 'Features/Laravel8') && version_compare(app()->version(), '8.0.0') < 0) {
```
Comment (canvural) [inline tests/FeaturesTest.php:41]: I think you can do `version_compare(app()->version(), '8.0.0', '<')` here

Comment (ThibaudDauce) [reply tests/FeaturesTest.php:41]: Yes but I thought it was less readable but as you wish

Comment (canvural) [reply tests/FeaturesTest.php]: IMO, `version_compare(...) < 0` is more confusing. Why do we expect it to be less than 0? (If you don't know the return types of version_compare) :smile:

**Review by canvural — APPROVED** [review]
Looks good! When the checks are passing you can remove the dummy test, and push again. Then we can merge! :+1:

**Comment by canvural** [comment]
Thank you!

File: tests/FeaturesTest.php (hunk 3 above)
Comment (szepeviktor) [reply tests/FeaturesTest.php]: Most parts of PHP is confusing!

**Pull request merged**